### PR TITLE
Update codeowners file for Azure Template

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Update CODEOWNERS to add label and reviewers for Azure Template changes.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -63,6 +63,9 @@
 # PRLabel: %Storage
 /sdk/storage/                                       @amishra-dev @rickle-msft @jaschrep-msft @gapra-msft @alzimmermsft @kasobol-msft
 
+# PRLabel: %EngSys
+/sdk/template/                                      @Azure/azure-sdk-eng
```
Comment (scbedd) [inline .github/CODEOWNERS:67]: Same comment on the other template. We don't support a team right now.

Comment (joshfree) [reply .github/CODEOWNERS:67]: @scbedd is this on the backlog for later this year?

Comment (chidozieononiwu) [reply .github/CODEOWNERS:67]: @joshfree Yes, and also I have a PR to implement that feature. https://github.com/Azure/azure-sdk-tools/pull/981

Comment (chidozieononiwu) [reply .github/CODEOWNERS:67]: Please can you go ahead and approve this. It should be fine for the automatic version increment. Can you also review https://github.com/Azure/azure-sdk-tools/pull/981 where I updated code owners to detect ID vs slug

**Review by weshaggard — APPROVED** [review]

**Pull request merged**